src/core/DropdownMenu/index.tsx

```diff
@@ -220,29 +237,34 @@ const DropdownMenu = <
     option: T,
     { selected }: AutocompleteRenderOptionState
   ) {
-    return (
-      <li {...optionProps}>
-        <StyledMenuItem
-          {...{ component: "div" }}
-          isMultiSelect={multiple}
-          selected={selected}
-          count={option.count}
-        >
-          <div>
-            {option.name}
-
-            {option.details && (
-              <StyledMenuItemDetails>{option.details}</StyledMenuItemDetails>
-            )}
-          </div>
+    let MenuItemContent;
 
-          {option.count && (
-            <StyledMenuItemCount className="menuItem-count">
-              {option.count}
-            </StyledMenuItemCount>
+    if (option.component) {
+      MenuItemContent = option.component;
+    } else {
+      MenuItemContent = (
+        <StyledMenuItemText>
+          {option.name}
+          {option.details && (
+            <StyledMenuItemDetails className="menuItem-details">
+              {option.details}
+            </StyledMenuItemDetails>
           )}
-        </StyledMenuItem>
-      </li>
+        </StyledMenuItemText>
+      );
+    }
+
+    return (
+      <MenuItem
+        column={option.count}
```

Comment:
nit: should we destruct `option`?

e.g., 

```ts
const { count, sdsIcon, ... } = option;

<MenuItem 
  column={count}
  ...
/>
```

Reply:
Yes, absolutely. I missed this one. 🫠